airbyte-featureflag/src/main/kotlin/Flags.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.featureflag
+
+/**
+ * If enabled, all messages from the source to the destination will be logged in 1 second intervals.
+ *
+ * This is a permanent flag and would implement the [Flag] type once converted from an environment-variable.
+ */
+object LogConnectorMessages : EnvVar(envVar = "LOG_CONNECTOR_MESSAGES", team = Team.PLATFORM_MOVE)
+
+object StreamCapableState : EnvVar(envVar = "USE_STREAM_CAPABLE_STATE")
+object AutoDetectSchema : EnvVar(envVar = "AUTO_DETECT_SCHEMA", team = Team.PLATFORM_MOVE)
+object NeedStateValidation : EnvVar(envVar = "NEED_STATE_VALIDATION")
+object ApplyFieldSelection : EnvVar(envVar = "APPLY_FIELD_SELECTION")
+
+/**
+ * The team that created and is responsible for individual feature-flags.
+ * TODO: finalize the options here!
+ */
+enum class Team {
```

Comment:
I know I was the one who suggested this, but I wonder if this should be more of an "area" or "focus" than team, simply because we may end up with different concerns that are owned by the same team (thinking out loud here).  Ultimately, we are trying to achieve some sort of categorization or identification to make it easy for those reading the code to know who to talk to about what a flag is being used for and/or if it can be removed.  I don't hold this too strongly -- I just wonder that this won't mean anything to open source users, as they know nothing about our internal team structure.

Reply:
This is indeed very internal, at the same time, the current feature flagging approach is not really targeted towards open source users. Usage of those flags may change quickly and I don't think we plan on offering any guarantees in terms API stability.